SalesforceCommerceCloud/pwa-kit
# Dynamic enablePWAKitPrivateClient that is based on config

Previously, if you were using `Template Extensibility` feature, it was unnecessary to create whole `overrides/app/components/_app-config/index.jsx` if you only want to use `PWAKitPrivateClient`. 

# Description

Previously, if you were using `Template Extensibility` feature, it was unnecessary to create whole `overrides/app/components/_app-config/index.jsx` if you only wanted to use `PWAKitPrivateClient`. 
it is said that Projects generated from PWA Kit 3.5 and above will be configured to use an SLAS private client by default in the documentation, with this PR, if you are using `Template Extensibility`, you can configure everything via configuration

# Types of Changes


# Changes

Added a new parameter on `default.js` file to manage `PWAKitPrivateClient` feature and it passed to `_app-config` file

# How to Test-Drive This PR

- Create a new PWA-Kit project
- Create a private SLAS client and add/configure it for your SFCC instance
- Enable `enablePWAKitPrivateClient` by changing default `false` value
- See if it works

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


## Localization

- [x] Don't need localization change
